processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -525,15 +362,15 @@ public int size()
       return size;
     }
 
-    protected T _get(ByteBuffer copyValueBuffer, int startOffset, int endOffset)
+    T bufferedIndexedGet(ByteBuffer copyValueBuffer, int startOffset, int endOffset)
     {
       final int size = endOffset - startOffset;
-      if (startOffset == endOffset) {
+      lastReadSize = size;
+      if (size == 0) {
         return null;
       }
+      copyValueBuffer.clear();
```

Comment:
Are you sure that `clear()` is correct.  I don't think there's any guarantee that just resetting things will get you the previous state.